contracts/SpokePool.sol

```diff
@@ -884,12 +964,40 @@ abstract contract SpokePool is
         }
     }
 
+    function _updateCountFromFill(
+        uint256 startingFillAmount,
+        uint256 endingFillAmount,
+        uint256 totalFillAmount,
+        int64 realizedLPFeePct,
+        address token,
+        bool localRepayment,
+        bool useContractFunds
+    ) internal {
+        // If this is a slow fill or it's an initial 0-fill, do nothing, as these should not impact the count.
+        if (useContractFunds || endingFillAmount == 0) return;
+
+        // If this is the first fill and it's partial, assume the rest of the fill will be slow filled (refunded on this chain).
+        if (startingFillAmount == 0 && totalFillAmount - endingFillAmount > 0) {
+            fillCounter[token] += _computeAmountPostFees(totalFillAmount - endingFillAmount, realizedLPFeePct);
+        }
+
+        // If this is not the first fill, remove the partial fill that was previously assumed.
+        if (startingFillAmount != 0) {
```

Comment:
Maybe this is premature, but is there a gas optimisation by making these two if statements mutually exclusive? For example:

```
        if (startingFillAmount == 0) {
            if (totalFillAmount - endingFillAmount > 0) {
                fillCounter[token] += _computeAmountPostFees(totalFillAmount - endingFillAmount, realizedLPFeePct);
            }
        } else {
        // If this is not the first fill, remove the partial fill that was previously assumed (startingFillAmount > 0).
```